sqlglot/expressions.py

```diff
@@ -6621,14 +6622,20 @@ def func(name: str, *args, dialect: DialectType = None, **kwargs) -> Func:
 
     from sqlglot.dialects.dialect import Dialect
 
-    converted: t.List[Expression] = [maybe_parse(arg, dialect=dialect) for arg in args]
-    kwargs = {key: maybe_parse(value, dialect=dialect) for key, value in kwargs.items()}
+    converted: t.List[Expression] = [maybe_parse(arg, dialect=dialect, copy=copy) for arg in args]
+    kwargs = {key: maybe_parse(value, dialect=dialect, copy=copy) for key, value in kwargs.items()}
 
     parser = Dialect.get_or_raise(dialect)().parser()
     from_args_list = parser.FUNCTIONS.get(name.upper())
 
     if from_args_list:
-        function = from_args_list(converted) if converted else from_args_list.__self__(**kwargs)  # type: ignore
+        if converted:
+            function: Func = maybe_parse(
```

Comment:
i don't think this is the right approach, instead, get the FUNCTIONS from parser and call that so you don't have to call sql and reparse

Reply:
Hmm, isn't that what we already do though? It doesn't seem sufficient, I believe you have to hit the parser for e.g. functions that exist in `FUNCTION_PARSERS`, otherwise we won't use that "special parsing" logic and the AST could end up being corrupted (like in the case of `LOG`).

Reply:
i don't think you can safely call function_parsers because they don't necessarily expect commas, i think you can only use FUNCTIONS whichwould handle arg order correctly

Reply:
Yeah I didn't mean to call function parsers directly, I meant let the parser figure out what parsing method to use by calling `maybe_parse`. The issue with `FUNCTIONS` is that they don't have special logic for "weird" functions such as `LOG` etc to figure out the correct order. See for example the defn. of `LOG`: https://github.com/tobymao/sqlglot/blob/main/sqlglot/parser.py#L4566

Reply:
Essentially `FUNCTIONS` doesn't have all sufficient information to figure out how to route the `exp.func` arguments to `args`.

Reply:
I'm not sure I understand the suggested path to solving the issue at hand. Are you thinking about moving `_parse_logarithm` to `FUNCTIONS` instead of `FUNCTION_PARSERS`?